django_mongodb/compiler.py

```diff
@@ -153,9 +153,19 @@ def get_columns(self):
         columns = (
             self.get_default_columns(select_mask) if self.query.default_cols else self.query.select
         )
-        return tuple((column.target.column, column) for column in columns) + tuple(
-            self.query.annotation_select.items()
-        )
+        annotation_idx = 1
+        result = []
+        for column in columns:
+            if hasattr(column, "target"):
+                # column is a Col.
+                target = column.target.column
+            else:
+                # column is a Transform in values()/values_list() that needs a
+                # name for $proj.
+                target = f"__annotation{annotation_idx}"
```

Comment:
I've just changed this. I added an _ between the annotation and the number. But it does not matter, I will keep the one that goes in this PR.

Reply:
This version is consistent with Django's internal aliasing [`__col{index}`](https://github.com/django/django/blob/3ac0e43207b4a1b9327ba12612005c285fbff7ee/django/db/models/sql/query.py#L565).